nunaliit2-js/src/main/js/nunaliit2/n2.couchDbPerspective.js

```diff
@@ -871,25 +872,29 @@ var ModelCouchDbView = $n2.Class({
 				};
 				
 			} else {
-				// This is a removed document
-				removed.push(oldDoc);
+				if( oldDoc ){
+					// This is a removed document
+					removed.push(oldDoc);
+				};
 			};
 		};
 		
 		// Detect added documents
-		for(var docId in docInfoMap){
-			var oldDoc;
+		for(docId in docInfoMap){
```

Comment:
Is it possible to collapse these for loops into one?  Instead of iterating over the docInfoMap twice?

Reply:
I thought so as well, but then I noticed that the objects being iterated over are different. That's why they were separated in the code originally.